src/core/componentSystems/PhysicsSystem.js

```diff
@@ -49,36 +49,31 @@ export class PhysicsSystem extends MRSystem {
         }
     }
 
+    /**
+     * @function
+     * @description The per global scene event update call.  Based on the captured physics events for the frame, handles all items appropriately.
+     */
     eventUpdate = () => {
+        mrjsUtils.physics.world.step(mrjsUtils.physics.eventQueue);
+
         for (const entity of this.registry) {
             if (entity.physics?.body == null) {
                 continue;
             }
-            if (entity instanceof MRModel) {
-                this.updateSimpleBody(entity);
-            } else if (entity instanceof MRDivEntity) {
-                this.updateUIBody(entity);
-            }
+            this.updateBody(entity);
         }
+
+        this.updateDebugRenderer();
     };
 
     /**
      * @function
-     * @description The generic system update call. Based on the captured physics events for the frame, handles all items appropriately.
+     * @description The per-frame system update call. Based on the captured physics events for the frame, handles all items appropriately.
      * @param {number} deltaTime - given timestep to be used for any feature changes
      * @param {object} frame - given frame information to be used for any feature changes
      */
     update(deltaTime, frame) {
-        mrjsUtils.physics.world.step(mrjsUtils.physics.eventQueue);
-
-        for (const entity of this.registry) {
-            if (entity.physics?.body == null) {
-                continue;
-            }
-            this.updateBody(entity);
-        }
-
-        this.updateDebugRenderer();
+        this.eventUpdate();
```

Comment:
oof nah we can't do it this way. `eventUpdate` is only meant to be called on events, not per frame.